### PR TITLE
improve failure test coverage

### PR DESCRIPTION
**Pull request by dipak-pawar**

### CODE REVIEW

File: pkg/controller/installconfig/installconfig_controller_test.go [hunk 1]
```diff
@@ -146,6 +147,126 @@ func TestInstallConfigController(t *testing.T) {
...
+					return errors.New(errMsg)
+				}
+				return cl.Client.Get(ctx, key, obj)
+			}
+
+			// when
+			_, err := r.Reconcile(request)
+
+			// then
+			assert.Error(t, err)
```
Comment (xcoulon) [inline pkg/controller/installconfig/installconfig_controller_test.go]:
```suggestion
			require.Error(t, err)
```

File: pkg/controller/installconfig/installconfig_controller_test.go [hunk 2]
```diff
@@ -146,6 +147,126 @@ func TestInstallConfigController(t *testing.T) {
...
+			cl.MockCreate = func(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
+				if _, ok := obj.(*olmv1.OperatorGroup); ok {
+					return errors.New(errMsg)
+				}
+				return cl.Client.Create(ctx, obj, opts...)
+			}
+
+			// first reconcile for ns creation
+			result, err := r.Reconcile(request)
+			assert.NoError(t, err)
```
Comment (xcoulon) [inline pkg/controller/installconfig/installconfig_controller_test.go]:
```suggestion
			require.NoError(t, err)
```

File: pkg/controller/installconfig/installconfig_controller_test.go [hunk 3]
```diff
@@ -146,6 +147,126 @@ func TestInstallConfigController(t *testing.T) {
...
+			assert.True(t, result.Requeue)
+
+			// when
+			_, err = r.Reconcile(request)
+
+			// then
+			assert.Error(t, err)
```
Comment (xcoulon) [inline pkg/controller/installconfig/installconfig_controller_test.go]:
```suggestion
			require.Error(t, err)
```

File: pkg/controller/installconfig/installconfig_controller_test.go [hunk 4]
```diff
@@ -146,6 +147,126 @@ func TestInstallConfigController(t *testing.T) {
...
+			cl.MockCreate = func(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
+				if _, ok := obj.(*olmv1alpha1.Subscription); ok {
+					return errors.New(errMsg)
+				}
+				return cl.Client.Create(ctx, obj, opts...)
+			}
+
+			// first reconcile for ns creation
+			result, err := r.Reconcile(request)
+			assert.NoError(t, err)
```
Comment (xcoulon) [inline pkg/controller/installconfig/installconfig_controller_test.go]:
```suggestion
			require.NoError(t, err)
```

File: pkg/controller/installconfig/installconfig_controller_test.go [hunk 5]
```diff
@@ -146,6 +147,126 @@ func TestInstallConfigController(t *testing.T) {
...
+			assert.True(t, result.Requeue)
+
+			// second reconcile for og creation
+			result, err = r.Reconcile(request)
+			assert.NoError(t, err)
```
Comment (xcoulon) [inline pkg/controller/installconfig/installconfig_controller_test.go]:
```suggestion
			require.NoError(t, err)
```

File: pkg/controller/installconfig/installconfig_controller_test.go [hunk 6]
```diff
@@ -146,6 +147,126 @@ func TestInstallConfigController(t *testing.T) {
...
+			assert.True(t, result.Requeue)
+
+			// when
+			_, err = r.Reconcile(request)
+
+			// then
+			assert.Error(t, err)
```
Comment (xcoulon) [inline pkg/controller/installconfig/installconfig_controller_test.go]:
```suggestion
			require.Error(t, err)
```

File: pkg/controller/installconfig/installconfig_controller_test.go [hunk 7]
```diff
@@ -200,6 +321,28 @@ func TestCreateOperatorGroupForChe(t *testing.T) {
...
+		cl.MockCreate = func(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
+			return errors.New(errMsg)
+		}
+		cheOg := che.NewOperatorGroup(cheOperatorNs)
+
+		// when
+		ogCreated, err := r.ensureCheOperatorGroup(testLogger(), cheOperatorNs, installConfig)
+
+		//then
+		assert.EqualError(t, err, errMsg)
```
Comment (xcoulon) [inline pkg/controller/installconfig/installconfig_controller_test.go]:
```suggestion
		require.EqualError(t, err, errMsg)
```

File: pkg/controller/installconfig/installconfig_controller_test.go [hunk 8]
```diff
@@ -222,7 +365,29 @@ func TestCreateSubscriptionForChe(t *testing.T) {
...
+		cl.MockCreate = func(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
+			return errors.New(errMsg)
+		}
+		cheSub := che.NewSubscription(cheOperatorNs)
+
+		// when
+		subCreated, err := r.ensureCheSubscription(testLogger(), cheOperatorNs, installConfig)
+
+		// then
+		assert.EqualError(t, err, errMsg)
```
Comment (xcoulon) [inline pkg/controller/installconfig/installconfig_controller_test.go]:
```suggestion
		require.EqualError(t, err, errMsg)
```

File: pkg/controller/installconfig/installconfig_controller_test.go [hunk 9]
```diff
@@ -298,11 +463,32 @@ func TestCreateNamespaceForChe(t *testing.T) {
...
+		errMsg := "something went wrong while creating ns"
+		cl.MockCreate = func(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
+			return errors.New(errMsg)
+		}
+
+		// when
+		nsCreated, err := r.ensureCheNamespace(testLogger(), installConfig)
+
+		// then
+		assert.EqualError(t, err, errMsg)
```
Comment (xcoulon) [inline pkg/controller/installconfig/installconfig_controller_test.go]:
```suggestion
		require.EqualError(t, err, errMsg)
```

File: pkg/controller/installconfig/installconfig_controller_test.go [hunk 10]
```diff
@@ -146,6 +147,126 @@ func TestInstallConfigController(t *testing.T) {
...
+					return errors.New(errMsg)
+				}
+				return cl.Client.Get(ctx, key, obj)
+			}
+
+			// when
+			_, err := r.Reconcile(request)
+
+			// then
+			require.Error(t, err)
```
Comment (alexeykazakov) [inline pkg/controller/installconfig/installconfig_controller_test.go]: Check the error message.

File: pkg/controller/installconfig/installconfig_controller_test.go [hunk 11]
```diff
@@ -146,6 +147,126 @@ func TestInstallConfigController(t *testing.T) {
...
+			// first reconcile for ns creation
+			result, err := r.Reconcile(request)
+			require.NoError(t, err)
+			assert.True(t, result.Requeue)
+
+			// when
+			_, err = r.Reconcile(request)
+
+			// then
+			require.Error(t, err)
```
Comment (alexeykazakov) [inline pkg/controller/installconfig/installconfig_controller_test.go]: Check the error message.

File: pkg/controller/installconfig/installconfig_controller_test.go [hunk 12]
```diff
@@ -146,6 +147,126 @@ func TestInstallConfigController(t *testing.T) {
...
+			// second reconcile for og creation
+			result, err = r.Reconcile(request)
+			require.NoError(t, err)
+			assert.True(t, result.Requeue)
+
+			// when
+			_, err = r.Reconcile(request)
+
+			// then
+			assert.Error(t, err)
```
Comment (alexeykazakov) [inline pkg/controller/installconfig/installconfig_controller_test.go]: Check the error message.

**Pull request merged**